pandas/tests/reshape/concat/test_concat.py

```diff
@@ -698,3 +698,36 @@ def test_concat_posargs_deprecation():
         result = concat([df, df2], 0)
     expected = DataFrame([[1, 2, 3], [4, 5, 6]], index=["a", "b"])
     tm.assert_frame_equal(result, expected)
+
+
+def test_concat_retain_attrs_df():
+    # GH#41828
+    d = {"col1": [1, 2], "col2": [3, 4]}
+    df1 = DataFrame(data=d)
+    df1.attrs = {1: 1}
+    df2 = DataFrame(data=d)
+    df2.attrs = {1: 1}
+    df = concat([df1, df2])
+    assert df.attrs[1] == 1
```

Comment:
use assert_equal (u can also compare attrs if u want)

though should prob add a flag to do this in the asset_ routines (i don't think we have this)

Reply:
@jreback  Thank you for the review. I wonder if do you mean pandas._testing.assert_equal? It seems that pandas._testing.assert_equal cannot handle comparison between integers.